pkg/plugins/manager.go

```diff
@@ -0,0 +1,170 @@
+package plugins
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-plugin"
+	"github.com/treeverse/lakefs/pkg/logging"
+)
+
+var (
+	errPluginGroupNotFound = fmt.Errorf("unknown plugin group")
+	errPluginTypeNotFound  = fmt.Errorf("unknown plugin type")
+)
+
+// PluginType specifies the type of related plugins
+// For example, "diff" is a type of plugins that know how to perform a diff.
+type PluginType string
+
+// PluginName specifies the name of a plugin
+// For example, "delta" is a plugin.
+type PluginName string
+
+// PluginIdentity identifies the plugin implementation.
+type PluginIdentity struct {
+	Impl               plugin.Plugin
```

Comment:
* IIUC this is not the implementation, this is the plugin for the implementation.  The implementation itself lives in a separate binary.  Consider calling it `Plugin`.
* Given that it has known input and output types, why do we need to configure this field?  I would expect the _same_ `plugin.Plugin` to work for all executables -- and then I'd expect it to be called `plugin` (or be entirely missing, or plugin_ if package name `plugin` cannot be re-used).

Reply:
sometimes you're just right...